qlty-analysis/src/workspace_entries/matchers/plugin_specific_exclude_matcher.rs

```diff
@@ -0,0 +1,215 @@
+use crate::{utils::fs::path_to_string, WorkspaceEntry};
+use qlty_config::config::Exclude;
+use std::path::PathBuf;
+
+use super::WorkspaceEntryMatcher;
+
+#[derive(Debug)]
+pub struct PluginSpecificExcludeMatcher {
+    plugin_name: String,
+    excludes: Vec<Exclude>,
```

Comment:
My understanding is that the `Exclude` structs have plugin information on them already, so I wasn't expecting to see them alongside a `plugin` field like this.

This `PluginSpecificExcludeMatcher` seems a bit... specific :) I am wondering if there might be a more elegant way to compose this behavior. Ideally the code would feel like a clean composition. 

What do you think?

Reply:
So tying this with the other [comment](https://github.com/qltysh/qlty/pull/1954/files#r2086759578). 

From my understanding you can have two types of [excludes](https://github.com/qltysh/cloud/issues/5925). One with only `file_patterns` (will match regardless of plugin) and then targeting `file_patterns` for given set of plugins.

This specific matcher is for the second case, if plugin(s) are specified. Otherwise the generic `ExcludeGroupsMatcher` (formerly `IgnoreGroupsMatcher`) is used.

If memory serves correct previously we did not have anything that actually worked to filter the plugins even though the syntax was there which caused rebuilding/breaking down of ignores and overrides into excludes and triage.